File: tests/if/elim/if_elim.cpp
int main() {
  int t;
  int f;

  if (true) {
    t += 1;
  }

  if (false) {
    f -= 1;
  }

  if (1 > 0) {
    t += 1;
  } else {
    t -= 1;
  }

  if (1 < 0) {
    f += 1;
  } else {
    t -= 1;
  }

  // TODO: if (false) {}

  return 0;
}

File: tests/if/elim/if_elim.ml
open Optitrust
open Prelude


let _ = Run.script_cpp (fun _ ->
  (* TODO: static analysis and expected failures
  !! Trace.failure_expected (
    If_basic.elim_true [nbMulti; cIf ~then_:[cVar "f"] ()];
    If_basic.elim_false [nbMulti; cIf ~then_:[cVar "t"] ()];
  ); *)
  !! If_basic.elim_true [nbMulti; cIf ~then_:[cVar "t"] ()];
  !! If_basic.elim_false [nbMulti; cIf ~then_:[cVar "f"] ()];
)
